// @twurple/api/lib/api/helix/raids/HelixRaidApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HelixRaidApi = void 0;
const tslib_1 = require("tslib");
const common_1 = require("@twurple/common");
const BaseApi_1 = require("../../BaseApi");
const HelixRaid_1 = require("./HelixRaid");
/**
 * The Helix API methods that deal with raids.
 *
 * Can be accessed using `client.raids` on an {@ApiClient} instance.
 *
 * ## Example
 * ```ts
 * const api = new ApiClient(new StaticAuthProvider(clientId, accessToken));
 * const raid = await api.raids.startRaid('125328655', '61369223');
 * ```
 */
let HelixRaidApi = class HelixRaidApi extends BaseApi_1.BaseApi {
    /**
     * Initiate a raid from a live broadcaster to another live broadcaster.
     *
     * @param from The raiding broadcaster.
     * @param to The raid target.
     */
    async startRaid(from, to) {
        const result = await this._client.callApi({
            type: 'helix',
            url: 'raids',
            method: 'POST',
            query: {
                from_broadcaster_id: (0, common_1.extractUserId)(from),
                to_broadcaster_id: (0, common_1.extractUserId)(to)
            }
        });
        return new HelixRaid_1.HelixRaid(result.data[0]);
    }
    /**
     * Cancels an initiated raid.
     *
     * @param from The raiding broadcaster.
     */
    async cancelRaid(from) {
        await this._client.callApi({
            type: 'helix',
            url: 'raids',
            method: 'DELETE',
            query: {
                broadcaster_id: (0, common_1.extractUserId)(from)
            }
        });
    }
};
HelixRaidApi = tslib_1.__decorate([
    (0, common_1.rtfm)('api', 'HelixRaidApi')
], HelixRaidApi);
exports.HelixRaidApi = HelixRaidApi;
